node/src/main/java/io/airlift/node/NodeConfig.java

```diff
@@ -169,4 +164,21 @@ public NodeConfig setConfigSpec(String configSpec)
         this.configSpec = configSpec;
         return this;
     }
+
+    public AddressType getAddressType()
+    {
+        return addressType;
+    }
+
+    @Config("node.internal-address-type")
```

Comment:
> internal-address-lookup-method

Yea, i like it

> but why do we need hostname vs fqdn?

We need FQDN when we want to setup HTTPS communication. When using FQDN it is possible to generate certificate for the top level domain, and use the same certificate on all the nodes.